dask/array/routines.py

```diff
@@ -404,6 +405,71 @@ def ediff1d(ary, to_end=None, to_begin=None):
     return r
 
 
+def _gradient_kernel(f, grad_varargs, grad_kwargs):
+    return np.gradient(f, *grad_varargs, **grad_kwargs)
+
+
+@wraps(np.gradient)
+def gradient(f, *varargs, **kwargs):
+    f = asarray(f)
+
+    if not all([isinstance(e, Number) for e in varargs]):
+        raise NotImplementedError("Only numeric scalar spacings supported.")
```

Comment:
Is behavior on non-integral spacings defined?

Reply:
In NumPy Arrays (or non-uniform spacings) are supported. Opted not to do that here. Would rather see users request this functionality first.

Reply:
...or maybe you were referring non-discrete spacings, in which case the answer is yes for both.

Reply:
Should add there is a test for non-discrete spacings. :)